nemo_text_processing/g2p/modules.py

```diff
@@ -271,37 +277,52 @@ def __init__(
-                as characters. If the word has multiple entries and ignore_ambiguous_words is True, it will be returned as characters.
-            use_stresses (Optional[bool]): Whether or not to include the stress symbols (ˈ and ˌ).
-            set_graphemes_upper (Optional[bool]): Whether or not to convert all graphemes to uppercase (if not converted to phonemes).
-                You may want to set this if there is an overlap between grapheme/IPA symbols.
+            phoneme_dict (str, Path, or Dict): Path to file in CMUdict format or an IPA dict object with CMUdict-like
+                entries. For example,
+                a dictionary file: scripts/tts_dataset_files/ipa_cmudict-0.7b_nv22.06.txt;
+                a dictionary object: {..., "Wire": [["ˈ", "w", "a", "ɪ", "ɚ"], ["ˈ", "w", "a", "ɪ", "ɹ"]], ...}.
+            locale (str): Locale used to determine a locale-specific tokenization logic. Currently, it supports "en-US",
+                "de-DE", and "es-ES". Defaults to "en-US". Specify None if implementing custom logic for a new locale.
+            apply_to_oov_word (Callable): Function that deals with the out-of-vocabulary (OOV) words that do not exist
+                in the `phoneme_dict`.
+            ignore_ambiguous_words (bool): Whether to handle word via phoneme_dict with ambiguous phoneme sequences.
                 Defaults to True.
+            heteronyms (str, Path, List[str]): Path to file that includes heteronyms (one word entry per line), or a
+                list of words.
+            use_chars (bool): Whether to include chars/graphemes in the token list. It is True if `phoneme_probability`
+                is not None or if `apply_to_oov_word` function ever returns graphemes.
+            phoneme_probability (Optional[float]): The probability (0.0 <= ε <= 1.0) that is used to balance the action
+                that a word in a sentence is whether transliterated into a sequence of phonemes, or kept as a sequence
+                of graphemes. If a random number for a word is greater than ε, then the word is kept as graphemes;
+                otherwise, the word is transliterated as phonemes. Defaults to None which is equivalent to setting it
+                to 1.0, meaning always transliterating the word into phonemes. Note that this code path is only run if
+                the word can be transliterated into phonemes, otherwise, if a word does not have an entry in the g2p
+                dict, it will be kept as graphemes. If a word has multiple pronunciations as shown in the g2p dict and
+                `ignore_ambiguous_words` is True, it will be kept as graphemes as well.
+            use_stresses (Optional[bool]): Whether to include the stress symbols (ˈ and ˌ).
+            grapheme_case (Optional[str]): Trigger converting all graphemes to uppercase, lowercase, or mix-cases.
```

Comment:
the words in the dictionary and the words in the original text both are converted into the same case, either upper or lower. It indicates that the processed words with `grapheme_case` in the dict and original text should be case-matched. While for the mixed-case option, the word in the original text should not change, while we need a mixed-case word in dict in order to map its pronunciation accordingly. **It is required to have mixed-case words when preparing dictionaries**, but I did not enforce it for now. 

So to answer your question, if `grapheme_case="mixed"`, and if words in the dictionary are all upper-cased. Then g2p will not find corresponding pronunciations because "Query" will not match "QUERY q w eh r iy".